scripts/test_wheel.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+set -e
+
+# The build image version that will be used for testing
+TC_BUILD_IMAGE_VERSION=1.0.7
+
+unknown_option() {
+  echo "Unknown option $1. Exiting."
+  exit 1
+}
+
+print_help() {
+  echo "Tests the wheel in the targets directory."
+  echo "Currently quite fragile: assumes there is exactly one wheel in the targets directory."
+  echo
+  echo "Usage: ./test_wheel.sh [options]"
+  echo
+  echo "  --docker-python2.7       Use docker to test on Python 2.7 in Ubuntu 12.04."
+  echo
+  echo "  --docker-python3.5       Use docker to test on Python 3.5 in Ubuntu 14.04."
+  echo
+  echo "  --docker-python3.6       Use docker to test on Python 3.6 in Ubuntu 18.04."
+  echo
+  exit 1
+} # end of print help
+
+# command flag options
+# Parse command line configure flags ------------------------------------------
+while [ $# -gt 0 ]
+  do case $1 in
+    --docker-python2.7)     USE_DOCKER=1;DOCKER_PYTHON=2.7;;
+    --docker-python3.5)     USE_DOCKER=1;DOCKER_PYTHON=3.5;;
+    --docker-python3.6)     USE_DOCKER=1;DOCKER_PYTHON=3.6;;
+    --help)                 print_help ;;
+    *) unknown_option $1 ;;
+  esac
+  shift
+done
+
+set -x
+
+SCRIPT_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+WORKSPACE=${SCRIPT_DIR}/..
+cd ${WORKSPACE}
+
+# If we are going to run in Docker,
+# send this command into Docker and bail out of here when done.
+if [[ -n "${USE_DOCKER}" ]]; then
+  # create the build and test images
+  # (this should ideally be a no-op if the image exists and is current)
+  $WORKSPACE/scripts/create_docker_images.sh
+
+  # Run the tests inside Docker
+  if [[ "${DOCKER_PYTHON}" == "2.7" ]] || [[ "${DOCKER_PYTHON}" == "3.5" ]]; then
+    docker run --rm -m=8g \
+      --mount type=bind,source=$WORKSPACE,target=/build,consistency=delegated \
+      -e "VIRTUALENV=virtualenv --python=python${DOCKER_PYTHON}" \
+      turicreate/build-image-14.04:${TC_BUILD_IMAGE_VERSION} \
+      /build/scripts/test_wheel.sh
+  elif [[ "${DOCKER_PYTHON}" == "3.6" ]]; then
+    docker run --rm -m=8g \
+      --mount type=bind,source=$WORKSPACE,target=/build,consistency=delegated \
+      -e "VIRTUALENV=virtualenv --python=python${DOCKER_PYTHON}" \
+      turicreate/build-image-18.04:${TC_BUILD_IMAGE_VERSION} \
+      /build/scripts/test_wheel.sh
+  fi
+
+  exit 0
```

Comment:
Seems like this should be an error.

Reply:
This is not an error; this is an early return for a happy path. If any of the above commands fails, the whole script will return a non-zero exit code (because of `set -e`).